build_tools/benchmarks/reporting/parse_shark_benchmarks.py

```diff
@@ -0,0 +1,308 @@
+                  str(len(iree_results)))
+            print(str(iree_results))
+            continue
+
+          baseline_results = df_baseline.loc[(df_baseline.model == model) &
+                                             (df_baseline.dialect == dialect) &
+                                             (df_baseline.data_type
+                                              == data_type) &
+                                             (df_baseline.device == device)]
+
+          if baseline_results.empty:
+            # We use snapshots of latencies for baseline. If it is a new
+            # benchmark that is not included in the snapshot yet, emit a
+            # warning.
+            print(
+                "Warning: No baseline results found for " + model + ", " +
+                dialect + ", " + data_type + ", " + device +
+                ". Using IREE version as baseline. Please update baseline csv.")
+            engine = iree_results.engine.iloc[0]
+            baseline_latency = iree_results.loc[iree_results.engine == engine]
+            baseline_latency = baseline_latency.iloc[0]["ms/iter"]
+          else:
+            engine = baseline_results.engine.iloc[0]
+            baseline_latency = baseline_results.loc[baseline_results.engine ==
+                                                    engine]
+            baseline_latency = baseline_latency.iloc[0]["ms/iter"]
+
+          iree_latency = iree_results.loc[iree_results.engine == "shark_iree_c"]
+          iree_latency = iree_latency.iloc[0]["ms/iter"]
+          iree_vs_baseline = html_utils.format_latency_comparison(
+              iree_latency, baseline_latency)
+
+          if df_shark is not None:
+            shark_results = df_shark.loc[(df_shark.model == model) &
+                                         (df_shark.dialect == dialect) &
+                                         (df_shark.data_type == data_type) &
+                                         (df_shark.device == device)]
+
+            shark_latency = shark_results.loc[shark_results.engine ==
+                                              "shark_iree_c"]
+            shark_latency = shark_latency.iloc[0]["ms/iter"]
+            shark_vs_baseline = html_utils.format_latency_comparison(
+                shark_latency, baseline_latency)
+            iree_vs_shark = html_utils.format_latency_comparison(
+                iree_latency, shark_latency)
+          else:
+            # If there are no SHARK benchmarks available, use default values.
+            # These columns will be hidden later.
+            shark_latency = 0
+            shark_vs_baseline = "1.0x faster"
+            iree_vs_shark = "1.0x faster"
+
+          summary.loc[len(summary)] = [
+              model, engine, data_type, dialect, device,
+              "{:.1f}".format(baseline_latency), "{:.1f}".format(iree_latency),
+              "{:.1f}".format(shark_latency), iree_vs_baseline,
```

Comment:
Can also use f strings here if you want

```suggestion
              f"{baseline_latency:.1f}", f"{iree_latency:.1f}",
              f"{shark_latency:.1f}", iree_vs_baseline,
```

Reply:
Done